### ISSUE TITLE
Performance: long loading times

### ISSUE THREAD

**Issue by practical-solutions** [ConX/dokuwiki-plugin-authorstats#45]
On a dokuwiki installation with about 2000 pages and 70 authors, the speed when saving pages and logging kept on growing, so it took 2-3 minutes in the end. The last_changes list was set to 7000. Deactivitang the plugin made instant saving and logging in possible again. Thus there seems to be some sort of performance issues when the last changes value is high.

**Comment by ConX** [comment]
Could you provide more details on the version of the wiki that this issue is referring to? Thank you!

**Comment by james2742** [comment]
Noticed this issue using linuxserver's docker image: https://hub.docker.com/r/linuxserver/dokuwiki

Slow login, save/cancel hangs when trying to exit the text editor.

Version:
```
DokuWiki version: Release 2020-07-29 "Hogfather"
PHP version 7.4.15
More than 32MB RAM (128 MB) available.
Changelog is writable
conf directory is writable
mb_string extension is available and will be used
Valid locale C.UTF-8;C;C;C;C;C found.
Debugging support is disabled
..
Your current permission for this page is 255
The current page is writable by the webserver
The current page is writable by you
The search index seems to be working
Server time seems to be okay. Diff: 0s
```

Haven't been able to find a fix other than switching to an older release, performance is as expected on bitnami's docker image: https://hub.docker.com/r/bitnami/dokuwiki/

Version:

```
DokuWiki version: Release 2018-04-22c "Greebo"
PHP version 7.3.20
More than 32MB RAM (268435456 bytes) available.
Changelog is writable
conf directory is writable
mb_string extension is available and will be used
Your locale C seems not to be a UTF-8 locale, you should fix this if you encounter problems.
Debugging support is disabled
..
Your current permission for this page is 255
The current page is writable by the webserver
The current page is writable by you
The search index seems to be working
Server time seems to be okay. Diff: 0s
```

**Comment by Digitalin** [comment]
Hi, 
I confirm the problem on a server with PHP 5.6 with Hogfather, a long time (between 50 seconds to 2 minutes) to log in and to save. It does not pose problem on navigation. It took me a while to discover witch plugin it was. Nothing relevant in the logs.

**Comment by ConX** [comment]
Hi @practical-solutions and @Digitalin, I know it has been a minute since I last worked on the plugin. I just started PR #52 in an attempt to fix any performance issues. Is it possible for you to install the plugin from source and test out the performance?

**Comment by Digitalin** [comment]
Hi @ConX,

With your fix, all seems ok for me (PHP 5.6 & PHP 7.4). Thanks a lot.

**Comment by ConX** [comment]
@Digitalin thanks for testing it!

**Comment by james2742** [comment]
> Hi @ConX,
> 
> With your fix, all seems ok for me (PHP 5.6 & PHP 7.4). Thanks a lot.

Same here, working on PHP 7.4 - Thanks @ConX
